chaoscenter/authentication/api/main.go

```diff
@@ -0,0 +1,180 @@
+	JwtSecret     string `required:"true" split_words:"true"`
+	AdminUsername string `required:"true" split_words:"true"`
+	AdminPassword string `required:"true" split_words:"true"`
+	DbServer      string `required:"true" split_words:"true"`
+	DbUser        string `required:"true" split_words:"true"`
+	DbPassword    string `required:"true" split_words:"true"`
+}
+
+func init() {
+	printVersion()
+
+	var c Config
+
+	err := envconfig.Process("", &c)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func main() {
+	// send logs to stderr so we can use 'kubectl logs'
+	_ = flag.Set("logtostderr", "true")
+	_ = flag.Set("v", "3")
+
+	flag.Parse()
+
+	client, err := utils.MongoConnection()
+	if err != nil {
+		log.Fatal("database connection error $s", err)
+	}
+
+	db := client.Database(utils.DBName)
+
+	// Creating User Collection
+	err = utils.CreateCollection(utils.UserCollection, db)
+	if err != nil {
+		log.Fatalf("failed to create collection  %s", err)
+	}
+
+	err = utils.CreateIndex(utils.UserCollection, utils.UsernameField, db)
+	if err != nil {
+		log.Fatalf("failed to create index  %s", err)
+	}
+
+	// Creating Project Collection
+	err = utils.CreateCollection(utils.ProjectCollection, db)
+	if err != nil {
+		log.Fatalf("failed to create collection  %s", err)
+	}
+
+	userCollection := db.Collection(utils.UserCollection)
+	userRepo := user.NewRepo(userCollection)
+
+	projectCollection := db.Collection(utils.ProjectCollection)
+	projectRepo := project.NewRepo(projectCollection)
+
+	miscRepo := misc.NewRepo(db, client)
+
+	applicationService := services.NewService(userRepo, projectRepo, miscRepo, db)
+
+	validatedAdminSetup(applicationService)
+
+	go runGrpcServer(applicationService)
+	runRestServer(applicationService)
+}
+
+func validatedAdminSetup(service services.ApplicationService) {
+	configs := map[string]string{"ADMIN_PASSWORD": utils.AdminPassword, "ADMIN_USERNAME": utils.AdminName, "DB_USER": utils.DBUser, "DB_SERVER": utils.DBUrl, "DB_NAME": utils.DBName, "DB_PASSWORD": utils.DBPassword, "JWT_SECRET": utils.JwtSecret}
+	for configName, configValue := range configs {
+		if configValue == "" {
+			log.Fatalf("Config %s has not been set", configName)
+		}
+	}
+
+	// Assigning UID to admin
+	uID := uuid.Must(uuid.NewRandom()).String()
+
+	// Generating password hash
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(utils.AdminPassword), utils.PasswordEncryptionCost)
+	if err != nil {
+		log.Println("Error generating password for admin")
+	}
+	password := string(hashedPassword)
+
+	adminUser := entities.User{
+		ID:       uID,
+		UserName: utils.AdminName,
+		Password: password,
+		Role:     entities.RoleAdmin,
+		Audit: entities.Audit{
+			CreatedAt: time.Now().Unix(),
+			UpdatedAt: time.Now().Unix(),
+		},
+	}
+
+	_, err = service.CreateUser(&adminUser)
+	if err == utils.ErrUserExists {
+		log.Println("Admin already exists in the database, not creating a new admin")
+	}
+	if err != nil && err != utils.ErrUserExists {
```

Comment:
```suggestion
	else if err != nil {
```
We already check equality in line 126. So again checking inequality is not required, it won't be equal for sure.